consumer/src/tree.rs

```diff
@@ -301,7 +392,28 @@ mod tests {
             tree: None,
             root: None,
         };
-        tree.update(second_update);
+        let mut got_updated_root_node = false;
+        let mut got_new_child_node = false;
+        tree.update_and_process_changes(second_update, |change| {
+            if let super::Change::NodeUpdated { old_data, new_node } = &change {
+                if new_node.id() == NODE_ID_1
+                    && old_data.children == Box::new([])
+                    && new_node.data().children == Box::new([NODE_ID_2])
+                {
+                    got_updated_root_node = true;
+                    return;
+                }
+            }
+            if let super::Change::NodeAdded(node) = &change {
+                if node.id() == NODE_ID_2 {
+                    got_new_child_node = true;
+                    return;
+                }
+            }
+            assert!(false);
```

Comment:
You could directly `panic!` with a message such as: Expected root node update and new child changes. At least it will be clear from the test results if it ever break.